platforms/jvm/language-jvm/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -482,6 +484,17 @@ public Property<Integer> getRelease() {
         return release;
     }
 
+    /**
+     * Configures whether Preview Features should be enabled for the compilation ({@code --enable-preview} compiler flag).
+     *
+     * @since 8.8
+     */
+    @Incubating
+    @Input
+    @Optional
```

Comment:
❓ Instead of `@Optional`, shouldn't this have a convention to set it to `false`?

Reply:
I think we could set the convention to `false` in the plugin, but I would keep `@Optional`. Since it's not an inherent property of compilation but a user-configurable switch, it's okay for it to be unset.

Also, it is a small breaking change (if that's a thing) to introduce a non-optional property on a class like that. If it was used for something else without being in conjunction with the plugin, then it would fail with the unset property.

I actually had the latter situation recently with the `Wrapper` task.

Reply:
Another reason for keeping it optional is the ability to detect the case when it is unconfigured by the user, in order to detect a clash with a corresponding CLI option.